tool/plugin/lib/plugin.dart

```diff
@@ -896,11 +887,11 @@ class DeployCommand extends ProductCommand {
   Future<int> doit() async {
     if (isReleaseMode) {
       if (!await performReleaseChecks(this)) {
-        return new Future(() => 1);
+        return 1;
```

Comment:
Not necessarily for this PR, but we should probably change this to returning a constant (like `cliExitError`).